src/SCRIPTS/BF/background.lua

```diff
@@ -40,18 +40,34 @@ local function run_bg()
             -- assuming when sensor value higher than 0 there is an telemetry connection
             -- only send datetime one time after telemetry connection became available
             -- or when connection is restored after e.g. lipo refresh
-            local now = getDateTime()
-            local year = now.year;
 
-            values = {}
-            values[1] = bit32.band(year, 0xFF)
-            year = bit32.rshift(year, 8)
-            values[2] = bit32.band(year, 0xFF)
-            values[3] = now.mon
-            values[4] = now.day
-            values[5] = now.hour
-            values[6] = now.min
-            values[7] = now.sec
+            if API_VERSION < 1.041 then
+                local now = getDateTime()
+                local year = now.year;
+
+                values = {}
+                values[1] = bit32.band(year, 0xFF)
+                year = bit32.rshift(year, 8)
+                values[2] = bit32.band(year, 0xFF)
+                values[3] = now.mon
+                values[4] = now.day
+                values[5] = now.hour
+                values[6] = now.min
+                values[7] = now.sec
+            else
+                local now = getRtcTime()
+
+                values = {}
+                values[1] = bit32.band(now, 0xFF)
+                now = bit32.rshift(now, 8)
+                values[2] = bit32.band(now, 0xFF)
+                now = bit32.rshift(now, 8)
+                values[3] = bit32.band(now, 0xFF)
+                now = bit32.rshift(now, 8)
+                values[4] = bit32.band(now, 0xFF)
+                values[5] = 0 -- we don't have milliseconds
```

Comment:
Is it worth documenting in here the expected payload, or is it documented on the FW side re: MSP_SET_RTC?